.ci/Jenkinsfile

```diff
@@ -0,0 +1,206 @@
+                string(name: 'P_CPU_LIMIT', value: cpuLimit),
+                string(name: 'P_MEM_LIMIT', value: memLimit),
+                string(name: 'P_TIMEOUT', value: '1800'),
+                string(name: 'PIP_EXTRA_DEPS', value: extraDeps),
+                string(name: 'PYTEST_COMMAND', value: './scripts/test.sh'),
+                string(name: 'PYTEST_CLI_ARGS', value: pytestCliArgs),
+                string(name: 'P_N_GPUS', value: nGpus),
+            ]
+        )
+    }
+}
+
+stage('Prepare') {
+    node (pCloud) {
+        // In order to print the env and get the commit url and hash, we must be on a real agent
+        sh 'printenv'
+
+        def loadedSCM = checkout scm
+
+        sh 'printenv'
+
+        gitUrl = loadedSCM.GIT_URL
+        gitBranch = loadedSCM.GIT_BRANCH
+        gitCommit = loadedSCM.GIT_COMMIT
+
+        echo "gitUrl: $gitUrl"
+        echo "gitBranch: $gitBranch"
+        echo "gitCommit: $gitCommit"
+
+        def jenkinsfileWorkspace = cloneJenkinsfilesRepo()
+
+        runWithChecks = load "$jenkinsfileWorkspace/utils/runWithChecks.groovy"
+        expandDockerMatrix = load "$jenkinsfileWorkspace/utils/expandDockerMatrix.groovy"
+        prChangeset = load "$jenkinsfileWorkspace/utils/prChangeset.groovy"
+
+        pytorchDockerChanged = prChangeset("docker/pytorch/")
+    }
+}
+
+def dockerImagePostBuild(stagingImageTag) {
+    if (gitBranch == "main") {
+        // no need to run tests again
+        return
+    }
+    parallel([
+        "$stagingImageTag - all": { ->
+            build(
+                job: 'scratch/pytest',
+                parameters: [
+                    string(name: 'P_CLOUD', value: pCloud),
+                    string(name: 'P_GIT_REPO', value: gitUrl),
+                    string(name: 'P_GIT_COMMIT', value: gitCommit),
+                    string(name: 'P_DOCKER_IMAGE', value: stagingImageTag),
+                    string(name: 'P_CPU_LIMIT', value: "2"),
+                    string(name: 'P_MEM_LIMIT', value: "7Gi"),
+                    string(name: 'P_TIMEOUT', value: '1800'),
+                    string(name: 'PIP_EXTRA_DEPS', value: 'all'),
+                    string(name: 'PYTEST_COMMAND', value: './scripts/test.sh'),
+                    string(name: 'PYTEST_CLI_ARGS', value: '--test_duration all'),
+                ]
+            )
+        }
+    ])
+}
+
+stage('Build') {
+    def jobs = [:]
+    if (pytorchDockerChanged) {
+        jobs << expandDockerMatrix(
+            P_CLOUD: pCloud,
+            P_BUILD_MATRIX: './composer/pytorch_build_matrix.sh',
+            P_BUILD_MATRIX_GIT_REPO: 'https://github.com/mosaicml/testing.git',  // TODO RJPP_SCM_URL
+            P_BUILD_MATRIX_GIT_COMMIT: 'main', // TODO RJPP_BRANCH
+            P_DOCKERFILE: 'Dockerfile',
+            P_BUILD_CONTEXT: './docker/pytorch',
+            P_GIT_REPO: gitUrl,
+            P_GIT_COMMIT: gitCommit,
+            P_CPU_LIMIT: '4',
+            P_MEM_LIMIT: '15Gi',
```

Comment:
Do we really need this amount of resources for the docker build? 4 cores and 15Gi of ram seems high

Reply:
I was getting sigkill errors on the build. We do have plenty of extra resources on the build agent so would prefer to leave as is.